contracts/ColonyFunding.sol

```diff
@@ -141,6 +142,89 @@ contract ColonyFunding is ColonyStorage, DSMath {
     return nonRewardPotsTotal[_token];
   }
 
+  function getGlobalRewardPayoutCount() public view returns (uint256) {
+    return globalRewardPayoutCount;
+  }
+
+  function getUserRewardPayoutCount(address _user) public view returns (uint256) {
+    return userRewardPayoutCount[_user];
+  }
+
+  function startNextRewardPayout(address _token) public auth returns(uint256) {
+    require(!activeRewardPayouts[_token]);
+
+    activeRewardPayouts[_token] = true;
+    globalRewardPayoutCount += 1;
+
+    //TODO: Lock everyones tokens
+
+    uint256 amount = pots[0].balance[_token];
+    pots[0].balance[_token] = 0;
+
+    rewardPayoutCycles[globalRewardPayoutCount] = RewardPayoutCycle(
+      IColonyNetwork(colonyNetworkAddress).getReputationRootHash(),
+      amount,
+      amount,
+      _token,
+      block.timestamp
+    );
+
+    return globalRewardPayoutCount;
+  }
+
+  function claimRewardPayout(uint8 _payoutId, int256 _userReputation, int256 _totalReputation) public {
+    require(block.timestamp - rewardPayoutCycles[_payoutId].blockTimestamp < 5184000000); // 60 days
+    require(_payoutId - userRewardPayoutCount[msg.sender] == 1);
+
+    //TODO: Prove that userReputation and totalReputation in reputationState are correct
+
+    uint256 userTokens = token.balanceOf(msg.sender);
+    uint256 totalTokens = sub(token.totalSupply(), token.balanceOf(address(this)));
+
+    require(userTokens > 0 && totalTokens > 0 && _userReputation > 0 && _totalReputation > 0);
+
+    uint256 diff = 76 - SafeMath.numDigits(totalTokens * uint256(_totalReputation));
```

Comment:
`uint256` can hold up to a 78 digit number in base 10, so this can underflow to a very large value, which I'm not terribly happy with. I *think* the worse case scenario there is that `reward` ends up being zero (i.e. someone gets less than they should, rather than more than they should), but users would be pretty annoyed about that. Did this implementation come from anywhere in particular? Why did you choose 76?

Reply:
I thought it was 77 (idiot me), and 1 will be added in `SafeMath.percent`. Which is not good, anyway.
About the underflows and overflows i was not so sure, thats why i didn't added the tests for more extreme values.
We can can continue this on Gitter, i just realized a few more things.